code/game/turfs/simulated/floor/plating.dm

```diff
@@ -232,6 +234,56 @@
 				CW.update_icon(0)
 		playsound(src, 'sound/items/Screwdriver.ogg', 80, 1)
 
+/turf/simulated/floor/plating/metalfoam
+	name = "foamed metal plating"
+	icon_state = "metalfoam"
+	var/metal = MFOAM_ALUMINUM
+
+/turf/simulated/floor/plating/metalfoam/iron
+	icon_state = "ironfoam"
+	metal = MFOAM_IRON
+
+/turf/simulated/floor/plating/metalfoam/update_icon()
+	switch(metal)
+		if(MFOAM_ALUMINUM)
+			icon_state = "metalfoam"
+		if(MFOAM_IRON)
+			icon_state = "ironfoam"
+
+/turf/simulated/floor/plating/metalfoam/attackby(var/obj/item/C, mob/user, params)
+	if(..())
+		return 1
+	else if(istype(C) && C.force)
```

Comment:
yes
but it doesn't need to be an else
because the if block above
will return if true